main.js

```diff
@@ -149,12 +149,14 @@ module.exports = function (argumentOptions) {
         connectionOptions.tls = options.tls;
     }
 
-    if (options.routes) {
-        connectionOptions.routes = options.routes;
+    if (options.connections.routes) {
+        console.log("----- received and setting a route config from calling application");
+        server.settings.connections.routes.files = options.connections.routes.files;
     }
 
-    if (options.state) {
-        connectionOptions.state = options.state;
+    if (options.connections.routes.state) {
+        console.log("----- received and setting a state config from calling application");
+        server.settings.connections.routes.state = options.connections.routes.state;
     }
```

Comment:
Yeah I was talking with Sam about doing the check for the object property as it may not be passed in.  

I typically follow the thinking laid out here:
http://davidbcalhoun.com/2011/checking-for-undefined-null-and-empty-variables-in-javascript/

I didn't see any other object property checking in the main.js file - is there a standard that CNN follows?

I would usually put something along the lines of:
if(typeof options.connections !== "undefined") but since we are on a nested object that could have one or more parent/child keys undefined I didn't know if there was a standard utility function already being used to check for the key/value existence?
```
has = function(obj, key) {

    return key.split(".").every(function(x) {

        if(typeof obj != "object" || obj === null || ! x in obj)

            return false;

        obj = obj[x];

        return true;
    });
}
```

or something to crawl the object:  

```
var lookup= function (obj, key) {
    var type = typeof key, i = 0, length;
    if (type == "string" || type == "number") {
      key = ("" + key).replace(/\[(.*?)\]/g, function (m, key) { //handle case where [1] or ['xa'] may occur
          return "." + key.replace(/^["']|["']$/g, ""); //strip quotes at the start or end of the key
      }).split(".");
    }
    for (length = key.length; i < length; i++) {
      if (_.has(obj, key[i])) obj = obj[key[i]];
      else return void 0;
    }
    return obj;
  }

var options = {};
options.connections = {};
console.log(lookup(options, "connections"));
```